pkg/operator/configobservation/scheduler/observe_scheduler.go

```diff
@@ -36,7 +39,12 @@ func ObserveDefaultNodeSelector(genericListers configobserver.Listers, recorder
 	if err != nil {
 		return prevObservedConfig, errs
 	}
-
+	if schedulerConfig.Spec == (configv1.SchedulerSpec{}) {
```

Comment:
I prefer the defaulting pattern used in:

https://github.com/openshift/cluster-kube-apiserver-operator/blob/3568cbf5b403833349f73eb5de226ad60e10b5d5/pkg/operator/configobservation/auth/auth_metadata.go#L52